doozerlib/cli/__main__.py

```diff
@@ -2037,16 +2040,26 @@ def get_channel_versions(channel):
 
         return descending_versions, edges
 
+    # New rules, source an OTA managed datsource that specifies minor_floor, minor_block_list,
+    # z_floor, and z_block_list
+    url = f'{suggestions_url}/{major}.{minor}-{arch}.yaml'
+    req = urllib.request.Request(url)
+    req.add_header('Accept','application/yaml')
+    suggestions = yaml.safe_load(exectools.urlopen_assert(req))
     upgrade_from = set()
-    upgrade_from.update(get_channel_versions(prev_candidate_channel)[0][:2])
-    upgrade_from.update(get_channel_versions(prev_stable_channel)[0][:2])
+    prev_versions, prev_edges = get_channel_versions(prev_candidate_channel)
+    curr_versions, current_edges =get_channel_versions(candidate_channel)
+    for v in prev_versions:
+        if ( semver.VersionInfo.parse(v) >= semver.VersionInfo.parse(suggestions['minor_floor']) and
+            semver.VersionInfo.parse(v) <  semver.VersionInfo.parse(f'{major}.{minor}.0') and
```

Comment:
you probably don't need this filter, because you are looking at `prev_candidate_channel`.  But I'm in favor of keeping it, dropping `prev_candidate_channel` and parsing these out of `curr_versions` with this `minor`-cap.  Might have to do something to also exclude RCs and other pre-4.y.0 stuff...  Can we just look at the parsed minor value?